modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java

```diff
@@ -2297,6 +2309,54 @@ public boolean isEncryptionEnabled() {
         return this;
     }
 
+    /**
```

Comment:
Disagree. DataRegion is all about memory shared between some caches. It has nothing to do with disk compression. 
Say we have two regions: 100GB for fast caches and 10GB for less important slow ones. This is unrelated to what must be compressed ant what not.

Reply:
DataRegionConfiguration in persistent mode is about how data from memory pages is moved to a disk - it has properties like pageEvictionMode and checkpointPageBufSize. For Cache diskPageCompression (and even just "disk page") is a too low-level concept.

Reply:
Ok, DataRegionConfiguration has some relation to persistence configuration. This is not really important here.

The important thing is that I do not see any reasons why caches with different compression levels/algorithms can not share the same data region. If we will create a separate data region for each page compression configuration, we may end up with ineffective memory utilization.

"Too low level concept" is a kind of unclear argument for me. Configuration can not be too high or too low level, it can only be either correctly or incorrectly placed with respect to granularity of it's effect. And my point is that having page compression configuration per data region is too coarse grained and restrictive.